usr/sbin/p11sak/p11sak.c

```diff
@@ -3660,10 +3848,45 @@ static int iterate_compare(const void *a, const void *b, void *private)
     return result;
 }
 
+static CK_BBOOL objclass_expected(CK_OBJECT_HANDLE obj, enum p11sak_objclass objclass)
+{
+    CK_OBJECT_CLASS class_val = 0;
+    CK_ATTRIBUTE attrs[] = {
+        { CKA_CLASS, &class_val, sizeof(class_val) },
```

Comment:
please no array here as well, its just one single attribute. 

Actually, here you can safely use get_attribute(), may save you a few lines here.

Reply:
Done.